src/libraries/Microsoft.PowerFx.Interpreter/StackDepthCounter.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    /// <summary>
+    /// This keeps track of how many calls deep we are in a function.
+    /// </summary>
+    internal struct StackDepthCounter
+    {
+        private readonly int _depth;
```

Comment:
>_depth [](http://example.com/codeflow?start=29&length=6)

nit: Rename to 
_remaining? 

This is counting down, not counting up.

Reply:
Also going to renaming the function calls to "Decrement" then instead of "Increment"

Reply:
Actually no i'm not.

Reply:
Done.